### PR TITLE
Fix some grammar in The Advanced Rust Programming Language

### PR DESCRIPTION
**Pull request by aij**
Just some grammar fixes and an assumed missing word.

r? @Gankro

### CODE REVIEW

**Comment by Gankro** [comment]
Wow, thanks!

File: src/doc/nomicon/atomics.md [hunk 1]
```diff
@@ -127,9 +127,9 @@ fundamentally unsynchronized and compilers are free to aggressively optimize
 them. In particular, data accesses are free to be reordered by the compiler on
 the assumption that the program is single-threaded. The hardware is also free to
 propagate the changes made in data accesses to other threads as lazily and
-inconsistently as it wants. Mostly critically, data accesses are how data races
+inconsistently as it wants. Most critically, data accesses are how data races
 happen. Data accesses are very friendly to the hardware and compiler, but as
-we've seen they offer *awful* semantics to try to write synchronized code with.
+we've seen they offer *awful* semantics with which to try to write synchronized code.
```
Comment (Gankro) [inline src/doc/nomicon/atomics.md]: While general grammar fixes are appreciated, this precise from of pedantry is up with which the Rust project will not put. :smile_cat:

**Comment by Gankro** [comment]
r=me with the single nit addressed.

File: src/doc/nomicon/atomics.md (hunk 1 above)
Comment (aij) [inline src/doc/nomicon/atomics.md]: I'm generally not that pedantic either. Reverted.

**Comment by Gankro** [comment]
LGTM, just need the commits to be squashed.

**Comment by aij** [comment]
Done

**Comment by Gankro** [comment]
@bors r+ rollup

Thanks!

**Pull request merged**